lib/utils.js

```diff
@@ -95,3 +95,62 @@ exports.findAvailablePort = function(app){
   });
 
 }
+
+exports.autoStoreData = function (req, res, next) {
+
+  // Middleware - store any data sent in session
+
+  if (!req.session.data){
+    req.session.data = {};
+  }
+
+  for (var i in req.body){
+    if (i.indexOf("_") != 0){
+      req.session.data[i] = req.body[i];
+    }
+  }
+
+  // send session data to all views
+
+  for (var i in req.session.data){
+    res.locals[i] = req.session.data[i];
+  }
+
+  next();
+
+}
+
+// Middleware - add variables that are available in all views
+
+exports.commonData = function(req, res, next){
+
+  res.locals.asset_path="/public/";
```

Comment:
I'm wary that we're defining the asset path in two places. Can it only be defined once?